wallet/wallet.go

```diff
@@ -3817,6 +3822,93 @@ func (w *Wallet) ImportScript(ctx context.Context, rs []byte) error {
 	return nil
 }
 
+// VotingXprivFromSeed derives a voting xpriv from a 32 byte seed.
+func (w *Wallet) VotingXprivFromSeed(seed []byte) (*hdkeychain.ExtendedKey, error) {
+	return votingXprivFromSeed(seed, w.ChainParams())
+}
+
+// votingXprivFromSeed derives a voting xpriv from a 32 byte seed. It returns
+// the private extended key at HD path m/votePurpose'.
+func votingXprivFromSeed(seed []byte, params *chaincfg.Params) (*hdkeychain.ExtendedKey, error) {
+	const (
+		op      errors.Op = "wallet.VotingXprivFromSeed"
+		seedLen           = 32
+	)
+
+	if len(seed) != seedLen {
+		return nil, errors.E(errors.Invalid, errors.Errorf("expected %d byte seed but got %d", seedLen, len(seed)))
+	}
+
+	master, err := hdkeychain.NewMaster(seed, params)
+	if err != nil {
+		errors.E(op, err)
+	}
+
+	purpose, err := master.Child(hdkeychain.HardenedKeyStart + votePurpose)
```

Comment:
I don`t think we should use a specific purpose for this, because it makes recovering using standard decred software harder.

**If** we're going to derive this inside the wallet (instead of just importing an xpriv without the wallet knowing _how_ the xpriv was originally generated), then I think we should make it so that standard dcrwalllet can completely recreate this, from the original seed.

Reply:
It's either import the seed or convert seed to xpriv in js in decrediton correct?

I thought this was better for a few reasons:
1. We have another way to differentiate this xpriv from account xprivs because it is a different depth.
2. We don't want anyone to use these addresses as payment addresses, so not being about to get to them from restoring from the seed was a plus imo.
3. It is not the 44' purpose, or what is specified in [bip 44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki), so we can make our own rules as to what is what. It seems we may need to do something like m'/purpose'/index and then take two children from that index, 0 and 1, use one for the submission script, sending the private key to the vspd, and the other for signatures. That way we never get out of sync. imo it's more confusing to say these are standard xprivs.

But I'll do what you think is best.

> because it makes recovering using standard decred software harder

What do you think is needed? Would an export function suffice? We could include working with this in another golang tool, as you were saying we need that anyway?

Reply:
> 1. We have another way to differentiate this xpriv from account xprivs because it is a different depth.

We don't really do this anywhere, since it would require adding more metadata to accounts inside dcrwallet. We only differentiate between native BIP0044 accounts and imported accounts and all imported accounts are the same.

> 2. We don't want anyone to use these addresses as payment addresses, so not being about to get to them from restoring from the seed was a plus imo.

Again, since we don't have additional metadata, after you've imported an account, there isn't anything preventing the user from deriving an address from it.

Reply:
Ok. Using the same logic as account xprivs.